src/pages/SignUp/index.tsx

```diff
@@ -0,0 +1,96 @@
+import { Link, useNavigate } from 'react-router-dom';
+import { SubmitHandler, useForm } from 'react-hook-form';
+import { yupResolver } from '@hookform/resolvers/yup';
+import { useSelector, useDispatch } from 'react-redux';
+import { signUpSchema } from '../../schemas';
+import {
+  Button, InputField, SelectInput, PhoneSection, Title,
+} from '../../components/common';
+import { genderOptions } from '../../constants/genderOptions';
+import { SignUpInputsType } from '../../types/userInputsTypes';
+import { signUp } from '../../state/actions/userActions';
+import { requestErrorsSelector, statusSelector } from '../../state/reducers/userReducer';
+import { RequestStatus } from '../../constants/requestStatus';
+
+function SignUp() {
+  const dispatch = useDispatch();
+  const navigate = useNavigate();
+  const status = useSelector(statusSelector);
+  const requestErrors = useSelector(requestErrorsSelector);
+
+  const isRequestLoading = status === RequestStatus.PENDING;
+
+  const errorsExist = Object.entries(requestErrors).length > 0;
```

Comment:
WDYT if we use a `useMemo` here? Since we want to prevent re-renders and have a dynamic and cached value of this expression this might be a good opportunity for this API.
```suggestion
  const isRequestLoading = useMemo(() => status === RequestStatus.PENDING, [status]);

  const errorsExist = useMemo(() =>  Object.entries(requestErrors).length > 0, [requestErrors]);
```